runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -1347,22 +1939,48 @@ TR::Register *J9::Power::TreeEvaluator::irdbariEvaluator(TR::Node *node, TR::Cod
    cg->stopUsingRegister(condReg);
 
    cg->machine()->setLinkRegisterKilled(true);
-
-   return objReg;
-#else
-   return NULL;
+   return objReg;   
 #endif
    }
 
 TR::Register *J9::Power::TreeEvaluator::ardbarEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
-   return TR::TreeEvaluator::ardbariEvaluator(node, cg);
+   /* rdbar & wrtbar nodes contain the sideEffect Child Node. 
+      During Field Watch we evaluate the sideEffect and it's used within the process.
+      The evaluation of the remaining children, as well as the load/store operation occur within the respective evaluator. */
+   TR::Node *sideEffectNode = node->getFirstChild();
+   TR::Register *sideEffectRegister = cg->evaluate(sideEffectNode);
+   if (cg->comp()->getOption(TR_EnableFieldWatch))
+      {
+      TR::TreeEvaluator::rdWrtbarHelperForFieldWatch(node, cg, sideEffectRegister, NULL);
+      }
+   cg->decReferenceCount(sideEffectNode);
+   return TR::TreeEvaluator::aloadEvaluator(node, cg);
```

Comment:
We used to call ardbariEvaluator here. Is it a correct change?

Reply:
Yes. In the past if CS was disabled we would just end up calling aloadEvaluator. CS never generates ardbar nodes, so we just call aloadEvaluator.